lib/utils/get_catergoric.dart

```diff
@@ -25,19 +25,29 @@
 
 library;
 
-// Group imports by dart, flutter, packages, local. Then alphabetically.
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import 'package:rattle/providers/stdout.dart';
 import 'package:rattle/providers/vars/types.dart';
+import 'package:rattle/r/extract_large_factors.dart';
 
-List<String> getCategoric(WidgetRef ref) {
+List<String> getCategoric(
+  WidgetRef ref,
+  bool removeIgnore,
+) {
   // The typesProvider lists the types for the different variables which we
   // need to know for parsing the R scripts.
 
   Map<String, Type> roles = ref.read(typesProvider);
 
+  String stdout = ref.watch(stdoutProvider);
+
+  List<String> largeFactors = extractLargeFactors(stdout);
+
   List<String> rtn = [];
   roles.forEach((key, value) {
-    if (value == Type.categoric) {
+    if (value == Type.categoric &&
```

Comment:
Could you explain the logic here. If I have a variable (like state) that is categoric and I choose to IGNORE it but it does not have largeFactors, then it gets listed for Group By? I think if I chose IGNORE I would not expect it to be listed as GROUP BY?

Reply:
Hi @gjwgit, code is updated.